rcl_action/src/rcl_action/types.c

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl_action/types.h"
+
+#include "rcl/error_handling.h"
+
+
+rcl_action_goal_info_t
+rcl_action_get_zero_initialized_goal_info(void)
+{
+  static rcl_action_goal_info_t goal_info = {
+    {0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u, 0u},
+    {0, 0u}
+  };
+  return goal_info;
+}
+
+rcl_action_goal_status_array_t
+rcl_action_get_zero_initialized_goal_status_array(void)
+{
+  static rcl_action_goal_status_array_t status_array;
+  return status_array;
+}
+
+rcl_action_cancel_request_t
+rcl_action_get_zero_initialized_cancel_request(void)
+{
+  static rcl_action_cancel_request_t request;
+  request.goal_info = rcl_action_get_zero_initialized_goal_info();
```

Comment:
@jacobperron this one's tricky, we rely on the compiler to be smart enough to not call `rcl_action_get_zero_initialized_goal_info()` every single time. How wrong would it be to go with `request = {0};` only?

Reply:
Fixed.